core/io/pread_spec.rb

```diff
@@ -26,11 +26,73 @@
       buffer.should == "567"
     end
 
+    it "shrinks the buffer in case of less bytes read" do
+      buffer = "foo"
+      @file.pread(1, 0, buffer)
+      buffer.should == "1"
+    end
+
+    it "grows the buffer in case of more bytes read" do
+      buffer = "foo"
+      @file.pread(5, 0, buffer)
+      buffer.should == "12345"
+    end
+
     it "does not advance the file pointer" do
       @file.pread(4, 0).should == "1234"
       @file.read.should == "1234567890"
     end
 
+    it "returns an empty string for maxlen = 0, regardless of the offset" do
+      @file.pread(0, 4).should == ""
+      @file.pread(0, 400).should == ""
+    end
+
+    it "does not reset the buffer when reading with maxlen = 0" do
+      buffer = "foo"
+      @file.pread(0, 4, buffer)
+      buffer.should == "foo"
+
+      @file.pread(0, 400, buffer)
+      buffer.should == "foo"
+    end
+
+    it "ignores the current offset" do
+      @file.pos = 3
+      @file.pread(4, 0).should == "1234"
+    end
+
+    it "converts maxlen and offset to Integers using #to_int" do
+      maxlen = mock('maxlen')
+      maxlen.should_receive(:to_int).and_return(4)
+      offset = mock('offset')
+      offset.should_receive(:to_int).and_return(0)
+      @file.pread(maxlen, offset).should == "1234"
+    end
+
+    it "converts a buffer to String using to_str (but you cannot use this value)" do
+      buffer = mock('buffer')
+      buffer.should_receive(:to_str).at_least(1).and_return("foo")
+      @file.pread(0, 4, buffer).should == "foo"
+      buffer.should_not.is_a?(String)
+      buffer.to_str.should == "foo"
+    end
```

Comment:
Both file content and a `#to_str` returned value are equal that seems confusing.

Reply:
I think I fixed the issue here. My test was supposed to show that if you read into a buffer using `#to_str`, it totally ignored the buffer. Turns out I started this test by copy-pasting the wrong code, and instead of reading 4 bytes from offset 0, it read 0 bytes from offset 4. This behaviour is already covered in "does not reset the buffer when reading with maxlen = 0".

I've updated the test, it all makes sense now.